node/src/Pos/Block/Logic/Header.hs

```diff
@@ -164,8 +178,14 @@ classifyHeaders inRecovery headers = do
     tipHeader <- DB.getTipHeader @ssc
     let tip = headerHash tipHeader
     haveOldestParent <- isJust <$> DB.blkGetHeader @ssc oldestParentHash
-    let headersValid = isVerSuccess $
-                       verifyHeaders (headers & _Wrapped %~ toList)
+    leaders <-
```

Comment:
Is it possible that we are classifying this headers before we verified genesis block for `oldestHeaderEpoch`? I can't proof that it's not. And if it's possible, then this line can hang, because it waits until LRC for `oldestHeaderEpoch` is finished.
If we don't hold `StateLock` here (I hope we don't) then I would recommend using `lrcSingleShot oldestHeaderEpoch` before calling `lrcActionOnEpochReason`. It should be harmless.